e2e/hook/config.go

```diff
@@ -0,0 +1,25 @@
+package hook
+
+import (
+	"os"
+
+	"k8s.io/client-go/rest"
+)
+
+const envVarUseInsecure string = "E2E_USE_INSECURE_TLS"
+
+func setTLSFromEnv(cfg *rest.Config) {
+	cfg.Insecure = os.Getenv(envVarUseInsecure) != "false"
+}
+
+type configMutationFunc func(*rest.Config)
+
+var mutationFuncs = []configMutationFunc{
+	setTLSFromEnv,
+}
+
+func ProcessConfig(cfg *rest.Config) {
```

Comment:
I'd say `MutateConfig` gives a better idea of what kind of processing it's doing. WDYT?

```suggestion
func MutateConfig(cfg *rest.Config) {
```

Reply:
Done in 9a83a18.